gensim/models/tfidfmodel.py

```diff
@@ -272,7 +272,7 @@ def __getitem__(self, bow, eps=1e-12):
 
         vector = [
             (termid, tf * self.idfs.get(termid))
-            for termid, tf in zip(termid_array, tf_array) if self.idfs.get(termid, 0.0) != 0.0
+            for termid, tf in zip(termid_array, tf_array) if self.idfs.get(termid, 0.0) > eps
```

Comment:
Comparing in absolute value safer (easier to read and comprehend, plus less room for error).

Reply:
I had a discussion with @menshikh-iv stating otherwise. @menshikh-iv I am reverting this.

Reply:
@piskvorky I did the same here, but you asked me to revert it.